pkg/skaffold/deploy/helm.go

```diff
@@ -70,30 +71,38 @@ func (h *HelmDeployer) Labels() map[string]string {
 }
 
 func (h *HelmDeployer) Deploy(ctx context.Context, out io.Writer, builds []build.Artifact, labellers []Labeller) error {
-	var dRes []Artifact
-
 	event.DeployInProgress()
-
-	for _, r := range h.Releases {
-		results, err := h.deployRelease(ctx, out, r, builds)
+	releases := make([]string, len(h.Releases))
+	for i, r := range h.Releases {
+		name, err := h.deployRelease(ctx, out, r, builds)
 		if err != nil {
-			releaseName, _ := evaluateReleaseName(r.Name)
-
 			event.DeployFailed(err)
-			return errors.Wrapf(err, "deploying %s", releaseName)
+			return errors.Wrapf(err, "deploying %s", name)
 		}
-
-		dRes = append(dRes, results...)
+		releases[i] = name
 	}
 
+	manifests := h.getManifestsFromReleases(ctx, releases)
+	manifests, err := manifests.SetLabels(merge(labellers...))
```

Comment:
This only patches the yaml manifest bytes. It does not yet apply the manifest. I also checked with the helm demo and no labels were applied.
I think you still need to do the live-patching which you removed below.

Reply:
That could be true. i should call the patch command like we do for kubectl and kustomize.

Reply:
Yes, that should work. Good idea!